vaadin-flow-components-shared-parent/vaadin-flow-components-base/src/main/java/com/vaadin/flow/component/shared/Tooltip.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright 2022 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.shared;
+
+import java.io.Serializable;
+
+import com.vaadin.flow.component.Component;
+import com.vaadin.flow.component.UI;
+import com.vaadin.flow.dom.Element;
+import com.vaadin.flow.function.SerializableRunnable;
+
+/**
+ * A handle that can be used to configure and control tooltips.
+ *
+ * @author Vaadin Ltd
+ */
+public class Tooltip implements Serializable {
+
+    /**
+     * The {@code <vaadin-tooltip>} element controlled by this tooltip instance.
+     */
+    private final Element tooltipElement = new Element("vaadin-tooltip");
+
+    /**
+     * Tooltip position in relation to the target element.
+     */
+    public enum TooltipPosition {
+        TOP_START("top-start"), TOP("top"), TOP_END("top-end"), BOTTOM_START(
+                "bottom-start"), BOTTOM("bottom"), BOTTOM_END(
```

Comment:
Should we use `//@formatter:off` here, as we do in theme variants enums?

Reply:
Done